### PR TITLE
Optimisation Fix - Get full batch size in one query

### PR DESCRIPTION
**Pull request by zegnus**
This PR replaces a full loop over all the rows retrieved by a query by one single query that does all the sums in the query itself

### CODE REVIEW

**Comment by friedger** [comment]
:+1:

**Comment by friedger** [comment]
I assume that all tests still pass :-)

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java [hunk 1]
```diff
@@ -84,18 +84,15 @@ long getBatchSizeInBytes(long batchId) {
...
                     null);
 
-            int totalBytesColumnIndex = cursor.getColumnIndexOrThrow(COLUMN_TOTAL_BYTES);
-
-            while (cursor.moveToNext()) {
-                int individualDownloadSize = cursor.getInt(totalBytesColumnIndex);
-                totalSize += individualDownloadSize;
-            }
+            cursor.moveToFirst();
+            totalSize = cursor.getLong(0);
```
Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java]: what is this column id? I thought you needed to define another column variable in the raw query and then read from that?

Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java]: It is the column index of the projection.


  [image: --]
Friedger Müffke
[image: http://]about.me/friedger
<http://about.me/friedger>


2015-07-01 12:00 GMT+02:00 Adam Brown <notifications@github.com>:

**Comment by blundell** [comment]
if you say *Optimisation* you beg the question where are your performance figures to show the before and after gain?

**Comment by xrigau** [comment]
lol @friedger don't think there are many tests here

**Comment by xrigau** [comment]
:+1: @blundell

**Comment by amlcurran** [comment]
:+1: It'd be really good if someone could do a H&T on advanced SQL stuff, I don't know any of this :)

**Comment by ronocod** [comment]
I don't know enough to do a H&T, some useful ones here though: https://www.sqlite.org/lang_aggfunc.html
and some less useful ones: https://www.sqlite.org/lang_corefunc.html

**Pull request merged**